src/main/java/uk/gov/hmcts/reform/finrem/caseorchestration/controllers/AmendApplicationContestedController.java

```diff
@@ -0,0 +1,126 @@
+    public ResponseEntity<AboutToStartOrSubmitCallbackResponse> amendApplicationAppSolicitor(
+        @RequestBody CallbackRequest callbackRequest,
+        @RequestHeader(value = AUTHORIZATION_HEADER, required = false) final String authToken
+    ) {
+        Map<String, Object> caseData = callbackRequest.getCaseDetails().getData();
+
+        List<String> errors = new ArrayList<>();
+        validateApplicantSolicitorPostcodeDetails(caseData, errors);
+
+        return ResponseEntity.ok(AboutToStartOrSubmitCallbackResponse.builder().data(caseData).errors(errors).build());
+    }
+
+    @PostMapping(path = "/amend-application-app", consumes = MediaType.APPLICATION_JSON_VALUE,
+        produces = MediaType.APPLICATION_JSON_VALUE)
+    @Operation(description = "Validate postcode on applicant details page")
+    public ResponseEntity<AboutToStartOrSubmitCallbackResponse> amendApplicationApp(
+        @RequestBody CallbackRequest callbackRequest,
+        @RequestHeader(value = AUTHORIZATION_HEADER, required = false) final String authToken
+    ) {
+        Map<String, Object> caseData = callbackRequest.getCaseDetails().getData();
+        List<String> errors = new ArrayList<>();
+        validateApplicantPostcodeDetails(caseData, errors);
+
+        return ResponseEntity.ok(AboutToStartOrSubmitCallbackResponse.builder().data(caseData).errors(errors).build());
+    }
+
+    @PostMapping(path = "/amend-application-res-sol", consumes = MediaType.APPLICATION_JSON_VALUE,
+        produces = MediaType.APPLICATION_JSON_VALUE)
+    @Operation(description = "Validate postcode on respondent solicitor page")
+    public ResponseEntity<AboutToStartOrSubmitCallbackResponse> amendApplicationRespondentSolicitor(
+        @RequestBody CallbackRequest callbackRequest,
+        @RequestHeader(value = AUTHORIZATION_HEADER, required = false) final String authToken
+    ) {
+        Map<String, Object> caseData = callbackRequest.getCaseDetails().getData();
+        List<String> errors = new ArrayList<>();
+        validateRespondentPostcodeDetails(caseData, errors);
+
+        return ResponseEntity.ok(AboutToStartOrSubmitCallbackResponse.builder().data(caseData).errors(errors).build());
+    }
+
+    private void validateApplicantSolicitorPostcodeDetails(Map<String, Object> caseData, List<String> errors) {
+
+        if (YES_VALUE.equals(caseData.get(APPLICANT_REPRESENTED))) {
+            Map<String, Object> applicantSolicitorAddress = (Map<String, Object>) caseData.get(CONTESTED_SOLICITOR_ADDRESS);
+            String applicantSolicitorPostCode = (String) applicantSolicitorAddress.get(postCode);
+
+            if (StringUtils.isBlank(applicantSolicitorPostCode)) {
+                errors.add("Postcode field is required for applicant solicitor address.");
+            }
+        }
+
+    }
+
+    private void validateApplicantPostcodeDetails(Map<String, Object> caseData, List<String> errors) {
+        String applicantPostCode = null;
```

Comment:
You could remove this initializer if you wanted; copy what you did in validateApplicantSolicitorPostcodeDetails.

Reply:
[26f67a8](https://github.com/hmcts/finrem-case-orchestration-service/pull/1775/commits/26f67a837783a0ebb5c7ff08b8b58c5b269e9228)